Codespell: config, action and typos fixed

yarikoptic opened the pull request:
I was nearby, so decided to "stop by" and say "Hi"

yarikoptic committed:
Add rudimentary codespell config

yarikoptic committed:
Add github action to codespell main on push and PRs

yarikoptic committed:
Ignore .csv files, although someone might want to check them up

matthew-brett commented on extra/mean_sq_deviations.md:
```diff
@@ -66,8 +66,8 @@ We can find this location by transforming the formula in {eq}`sse_c` above
 {eq}`sse_c` is equal to zero, it means the *gradient* of 
-{eq}`sse_c` is 0, and this is true when we are at a peak or a trough of
-{eq}`sse_c`.  We want the trough.
+{eq}`sse_c` is 0, and this is true when we are at a peak or a through of
```
Oops - "trough" is correct !

matthew-brett reviewed (commented):
Thanks much for this - but - any idea how to tell the thing not to second guess "trough" there?

yarikoptic committed:
Ignore rare but legit trough

yarikoptic commented:
Thanks for catching. I added it to skips in the config, and reran the fixup.

yarikoptic commented on permutation/population_permutation.Rmd:
```diff
@@ -80,10 +80,10 @@ Leave voters into another array, called `leave_ages`.
 # As above, we are just using Pandas here to get arrays for the
 # Remain and Leave voter ages.
-remainers = brexit_df[brexit_df['brexit_vote'] == 'Remain']
+remainders = brexit_df[brexit_df['brexit_vote'] == 'Remain']
```
but we should also ignore these `remainers` probably in all of their forms right?

yarikoptic replied:
I skipped them too now!

matthew-brett commented on permutation/population_permutation.Rmd:
```diff
@@ -80,10 +80,10 @@ Leave voters into another array, called `leave_ages`.
 # As above, we are just using Pandas here to get arrays for the
 # Remain and Leave voter ages.
-remainers = brexit_df[brexit_df['brexit_vote'] == 'Remain']
+remainders = brexit_df[brexit_df['brexit_vote'] == 'Remain']
```
Ah - sorry - also `remainers` is correct here.

yarikoptic replied:
yeap -- we saw it in parallel, fixed already, sorry for the sloppy original submission ;-)

yarikoptic committed:
we are so good to remainers

yarikoptic committed:
[DATALAD RUNCMD] run codespell throughout

=== Do not change lines below ===
{
 "chain": [],
 "cmd": "codespell -w",
 "exit": 0,
 "extra_inputs": [],
 "inputs": [],
 "outputs": [],
 "pwd": "."
}
^^^ Do not change lines above ^^^

matthew-brett reviewed (commented):
Oops - spotted another not-for-correction word

matthew-brett commented:
Excellent - thanks again.
